src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleApi.cs

```diff
@@ -306,31 +395,49 @@ public async Task<TestAnalyteGroup> GetOneTestGroupAsync(string testGroupId)
                 {
                     foreach (var testGroup in respContent.ApiResponse.Content.TestAnalyteGroups.Items)
                     {
-                        Event(null, new ClientApiLoggerEventArgs
-                        {
-                            EventLevel = EnumEventLevel.Trace,
-                            HttpStatusCode = respContent.ResponseMessage.StatusCode,
-                            ElapsedMs = watch.ElapsedMilliseconds,
-                            Module = "SampleApi",
-                            Message = $"GetOneTestGroupAsync Success"
-                        });
+                        Event(null, CreateLoggerArgs(EnumEventLevel.Trace, "GetOneTestGroupAsync Success", respContent.ResponseMessage.StatusCode, watch.ElapsedMilliseconds));
                         return testGroup;
                     }
                 }
 
-                Event(null, new ClientApiLoggerEventArgs
-                {
-                    EventLevel = EnumEventLevel.Warn,
-                    HttpStatusCode = respContent.ResponseMessage.StatusCode,
-                    ElapsedMs = watch.ElapsedMilliseconds,
-                    Module = "SampleApi",
-                    Message = $"GetOneTestGroupAsync Failed"
-                });
+                Event(null, CreateLoggerArgs(EnumEventLevel.Trace, "GetOneTestGroupAsync Failed", respContent.ResponseMessage.StatusCode, watch.ElapsedMilliseconds));
+
                 return null;
             }
             catch (Exception e)
             {
-                Event(e, new ClientApiLoggerEventArgs { EventLevel = EnumEventLevel.Error, Module = "SampleApi", Message = $"GetOneTestGroupAsync Failed - {e.Message}" });
+                Event(e, CreateLoggerArgs(EnumEventLevel.Error, $"GetOneTestGroupAsync Failed - {e.Message}"));
+                throw;
+            }
+        }
+
+        public async Task<bool> IsScheduledForUseAsync(string authTwinRefId, string entityType, string entityId)

```

Comment:
For the SDK would it be better to split this into two methods each with a hardcoded entityType?  Maybe IsAnalyteScheduledForUse and IsTestGroupScheduledForUse?